### PR TITLE
Print nice errors to the screen

### PR DESCRIPTION
**Pull request by whikloj**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2419

# What does this Pull Request do?

Captures the Exception thrown from SolrPhpClient and logs it to watchdog instead of printing it to the screen. Then it presents a simple text message.

# What's new?
There is a variable in the SolrPhpClient set based on the response from Solr, if Solr is shut down this displays a NOTICE. So I silenced that. Made a small function to help aggregate multiple error messages into a single message on screen and in the log. Also removed some extraneous try {} catch blocks (which I didn't realize where useless as the Exception was caught in the query_processor)

# How should this be tested?

Turn off your Solr and use Islandora, go to Search pages, etc.
See the large red Exception text.
Pull in this PR.
See the nice yellow warning text, check your log for the Exception text.

# Additional Notes:

Example:
* Does this change the interface, add a new feature, or otherwise change behaviours that would require updating documentation? unsure
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

File: SolrPhpClient/Apache/Solr/Service.php
Comment (DiegoPino) [inline SolrPhpClient/Apache/Solr/Service.php:337]: Wuhu! so you are editing the Client =) This opens a lot of possibilities ++

File: includes/query_processor.inc
Comment (DiegoPino) [inline includes/query_processor.inc:402]: Love this name.. we need a hook named the same way (kidding)

File: islandora_solr.module
Comment (DiegoPino) [inline islandora_solr.module]: Manidora?

**Review by DiegoPino — CHANGES_REQUESTED** [review]
Just the manidora piece. Rest looks good

File: SolrPhpClient/Apache/Solr/Service.php
Comment (whikloj) [reply SolrPhpClient/Apache/Solr/Service.php:337]: I wasn't going to, but then I saw that `islandora_solr_debug_mode` block above. Also I have given up replacing this client, easier to fix it in Islandora 8.

File: islandora_solr.module
Comment (whikloj) [reply islandora_solr.module]: DOH!

File: islandora_solr.module
Comment (DiegoPino) [inline islandora_solr.module]: Final request. Could we have a way of customizing this message? Since it's front facing.. i know i know we did not have it before so why now? Because people i know will ask for it for sure.. maybe a hook-ie? i mean, its not a blocker... since you can always disable reporting at all...but could be cool? sorry

**Review by DiegoPino — APPROVED** [review]
This looks great. What better reason to catch exceptions than actually letting them to be thrown! Good, good.

**Comment by whikloj** [comment]
**PHP 5.3.3** holding on to the buggy so your electric car moves very slowly.

**Comment by DiegoPino** [comment]
Restarting the test, error is of type "HttpConnectionException' with message" so not DCS... i guess

**Comment by whikloj** [comment]
@DiegoPino I've restarted it a couple times and it always seems to fail when setting up Islandora/Fedora

**Comment by DiegoPino** [comment]
If i was release manager i would say lets merge this anyway in 24 hours. I'm a bit done with travis....

**Comment by jonathangreen** [comment]
php 5 3 3

**Comment by whikloj** [comment]
Perhaps (😬) it is time (again) to consider shifting PHP 5.3.3 to an allowed failure.

**Comment by DiegoPino** [comment]
So.. seems still failing. Should i merge???

**Comment by whikloj** [comment]
@DiegoPino this is on the agenda for a TAG meeting on Friday, why don't we hold off until then.

**Comment by whikloj** [comment]
Closing in favour of #363

**Pull request closed without merging**
